app/src/main/java/com/moodup/movies/ui/home/HomeFragment.kt

```diff
@@ -36,36 +37,57 @@ class HomeFragment : Fragment() {
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         setUpSearchView()
-
-        linearLayoutManager = LinearLayoutManager(activity)
-        movies_recycler_view.layoutManager = linearLayoutManager
+        setUpRecyclerView()
 
         activity?.let {
             viewModel = ViewModelProvider(it).get(MovieViewModel::class.java)
         }
 
-        viewModel?.getMoviesResponseLiveData()?.observe(viewLifecycleOwner, Observer {
-            setUpAdapter(it)
+        viewModel?.movieLiveData?.observe(viewLifecycleOwner, Observer {
+            updateAdapter(it)
+        })
+
+        viewModel?.UIstateLiveData?.observe(viewLifecycleOwner, Observer { state ->
+            when (state) {
+                UIState.LOADING -> {
+                    showProgressBar()
+                }
+                UIState.ON_ERROR -> {
+                    showOnError()
+                }
+                UIState.ON_RESULT -> {
+                    hideProgressBar()
+                }
+                UIState.ON_EMPTY_RESULTS -> {
+                    showEmptyResults()
+                }
+            }
         })
+
+        viewModel?.getMovies("")
```

Comment:
Tutaj wpadło mi do głowy, że można dodać jeszcze jeden stan UIState.initialized i tam wywołać to.